pkg/disttask/framework/storage/table_test.go

```diff
@@ -125,6 +133,194 @@ func TestTaskTable(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, proto.TaskStateFailed, task.State)
 	require.ErrorContains(t, task.Error, "test error")
+
+	// succeed a pending task, no effect
+	id, err = gm.CreateTask(ctx, "key-success", "test", 4, []byte("test"))
+	require.NoError(t, err)
+	require.NoError(t, gm.SucceedTask(ctx, id))
+	task, err = gm.GetTaskByID(ctx, id)
+	require.NoError(t, err)
+	checkTaskStateStep(t, task, proto.TaskStatePending, proto.StepInit)
+	// succeed a running task
+	require.NoError(t, gm.SwitchTaskStep(ctx, task, proto.TaskStateRunning, proto.StepOne, nil))
+	task, err = gm.GetTaskByID(ctx, id)
+	require.NoError(t, err)
+	checkTaskStateStep(t, task, proto.TaskStateRunning, proto.StepOne)
+	startTime := time.Unix(time.Now().Unix(), 0)
+	require.NoError(t, gm.SucceedTask(ctx, id))
+	task, err = gm.GetTaskByID(ctx, id)
+	require.NoError(t, err)
+	checkTaskStateStep(t, task, proto.TaskStateSucceed, proto.StepDone)
+	require.GreaterOrEqual(t, task.StateUpdateTime, startTime)
+}
+
+func checkAfterSwitchStep(t *testing.T, startTime time.Time, task *proto.Task, subtasks []*proto.Subtask, step proto.Step) {
+	tm, err := storage.GetTaskManager()
+	require.NoError(t, err)
+	ctx := context.Background()
+	ctx = util.WithInternalSourceType(ctx, "table_test")
```

Comment:
use InitTableTest

Reply:
this test requires `store`, will change it later